### DISCUSSION TITLE
Autocompleting with `end` in ruby

### DISCUSSION THREAD

**Discussion by manuraj17** [zed-industries/zed#6515] [Feature Requests]
Hello,

I searched for this and failed to find a related discussion. 

Is there a way to enable auto completing `end` in ruby (and elixir) and such similar helpers? 

Eg: When we type `def` and add the method name and press enter the `end` to be automatically added.

**Comment by WarnerAbediR** [comment]
So, if I understand correctly, you're looking for a way to automatically add the "end" keyword when defining a method in Ruby and Elixir, right? Well, the good news is that both languages have tools to make your life easier!

In Ruby, there are several text editors and IDEs that support automatic insertion of the "end" keyword. For example, in Sub

As for Elixir, the language itself has a feature called "code snippets" that allows you to define custom snippets that can be automatically inserted into your code. There's even a library called "Alchemist" that provides a collection of code snippets for Elixir and its associated tools. You can use Alchemist to define your own snippets for adding "end" keywords and other frequently used code patterns.

**Reply by manuraj17** [reply]
@WarnerAbediR Sorry, I missed the reply. Looks like some of your reply is missed. Could you elaborate this part? 

> In Ruby, there are several text editors and IDEs that support automatic insertion of the "end" keyword. For example, in Sub

**Reply by clo4** [reply]
I don't know for sure if it is, but this strikes me as an AI generated response

**Reply by manuraj17** [reply]
@clo4 Fascinatingly - that's what my initial feel also was. A new user account too. Bit disappointed though if it is true; nevertheless - was hoping somebody would address it personally

**Comment by axsuul** [comment]
Anyone figure this out?

**Reply by cpjmcquillan** [reply]
I think you can use [Snippets](https://zed.dev/docs/snippets) for this.

Configuring this for Ruby adds `end` after `do`.

```json
{
  "do": {
    "prefix": "do",
    "body": "do\n\t$0\nend"
  }
}
```

**Reply by baim-a** [reply]
```json
 {
   "do": {
     "prefix": "do",
     "body": "do\n\t$0\nend",
     "description":  "Autocomplete 'do' block"
   }
 }
 ```
Thanks. Yet need add "description"

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
